### ISSUE TITLE
Build a CloudBigtableIO connector that works with Beam

### ISSUE THREAD

**Issue by sduskis** [GoogleCloudPlatform/cloud-bigtable-client#1009]

**Comment by ankurcha** [comment]
Any update on this?

**Comment by sduskis** [comment]
What breaks with 0.9.6?  We should probably take look at that first.  We'll take a look at updating CloudBigtableIO in the near future.  Q2 is coming, and we're looking at doing things like this.

**Comment by ankurcha** [comment]
Specifically it has to do with incompatible grpc/protobuf version and this segment of code:

```java
// Set data channel count to one because there is only 1 scanner in this session
      // Use retryOptionsToBuilder because absent in Bigtable library
      // TODO: replace with RetryOptions.toBuilder() when added to Bigtable library
      // Set batch size because of bug (incorrect initialization) in Bigtable library
      // TODO: remove setRetryOptions when fixed in Bigtable library
      BigtableOptions.Builder clonedBuilder = options.toBuilder()
          .setDataChannelCount(1)
          .setRetryOptions(
              retryOptionsToBuilder(retryOptions)
                  .setStreamingBatchSize(Math.min(retryOptions.getStreamingBatchSize(),
                      retryOptions.getStreamingBufferSize() / 2))
                  .build());
      BigtableOptions optionsWithAgent = clonedBuilder.setUserAgent(getUserAgent()).build();
```

**Comment by sduskis** [comment]
This is a cloud-bigtable issue, rather than a beam issue.

First, the plan is to fix that in beam: https://issues.apache.org/jira/browse/BEAM-1269

You should only use bigtable-hbase-dataflow, and not have any other explicit bigtable-* direct dependencies.

I'll try to add more context next week.

**Comment by ankurcha** [comment]
Thanks for the jira pointer. Ideally we want to use the bigtable-hbase-dataflow connector (which is the officially supported one by the dataflow team - i believe). We really wanted to use that but because that is still build against `com.google.cloud.dataflow`, we had to resort to using BigtableIO.

**Comment by sduskis** [comment]
I was thinking of a parallel bigtable-hbase-dataflow-v2 project... that's still maintained by the Cloud Bigtable team.

I'll spend some time next week to see how quick this task is.  Hopefully it should be quick.

**Comment by sduskis** [comment]
The Beam connector should live in the beam repository.

FYI, @dhalperi

**Comment by feczo** [comment]
Now that BigtableIO connection handling is improved are there still any caveats, pros/cons around which Bigtable client implementation one should use with Beam if hbase compatibility is not a need?

**Comment by ankurcha** [comment]
I have been using BigtableIO that ships with Beam for some time without any issues.

**Comment by sduskis** [comment]
We released 1.0.0-pre3 with a CloudBigtableIO Beam connector.  We're still working on examples and documentation.
